### PR TITLE
Docker cluster launcher

### PR DESCRIPTION
**Pull request by DmiSergeev**
* Creating `statemachine` docker image with SBT
* Launching clusters of docker containers
* 2 example cluster launchers (`counter` and `sqldb`)
* Launching cluster ReadMe (`docker/README.md`)
* State machine config changes
* State machine logging changes
* State machine now (temporarily) accepts both `signature of TX` and `signature of hash(TX)` in `Crypto.scala`

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+### Building docker image
+
+Locate to the Fluence root directory and run:
+
+```
+sbt docker
+```
+
+It would build a `statemachine/statemachine:latest` image.
```
Comment (alari) [inline docker/README.md]: Let's name it `fluencelabs/statemachine`.

File: statemachine/src/main/scala/fluence/statemachine/ServerRunner.scala [hunk 2]
```diff
@@ -40,7 +43,7 @@ import scala.language.higherKinds
  * according to Tendermint specification) and sends ABCI requests to `ABCIHandler`.
  */
 object ServerRunner extends IOApp with LazyLogging {
-  val DefaultABCIPoint: Int = 46658
+  val DefaultABCIPoint: Int = 26658
```
Comment (alari) [inline statemachine/src/main/scala/fluence/statemachine/ServerRunner.scala]: Is there any reason for this magic number?

File: statemachine/src/main/scala/fluence/statemachine/ServerRunner.scala [hunk 3]
```diff
@@ -116,10 +130,40 @@ object ServerRunner extends IOApp with LazyLogging {
...
+    config: StateMachineConfig
+  ): EitherT[F, StateMachineError, Seq[String]] =
+    EitherT.fromEither[F](
+      config.moduleFiles
+        .map(
+          name =>
+            Try({
+              val file = new File(name)
+              if (!file.exists())
+                throw new FileNotFoundException(name)
```
Comment (alari) [inline statemachine/src/main/scala/fluence/statemachine/ServerRunner.scala]: With `throw`, the stack trace is collected.

File: statemachine/src/main/scala/fluence/statemachine/error/StateMachineError.scala [hunk 4]
```diff
@@ -48,6 +48,15 @@ case class VmRuntimeError(override val code: String, override val message: Strin
 /**
  * Corresponds to errors occurred during State machine config loading.
  */
-case class ConfigLoadingError(override val code: String, override val message: String) extends StateMachineError {
+case class ConfigLoadingError(override val message: String) extends StateMachineError {
+  override val code: String = "ConfigLoadingError"
```
Comment (alari) [inline statemachine/src/main/scala/fluence/statemachine/error/StateMachineError.scala]: You may change `StateMachineEror` to an abstract class with a constructor, to remove noisy `override val` constructs.

File: statemachine/src/main/scala/fluence/statemachine/state/Committer.scala [hunk 5]
```diff
@@ -85,6 +85,7 @@ class Committer[F[_]](
   private def logState(state: MerkleTreeNode, height: Long): Unit = {
     logger.info("Commit: height={} hash={}", height, state.merkleHash.toHex)
     logger.debug("\n{}", state.dump())
+    logger.info("")
```
Comment (alari) [inline statemachine/src/main/scala/fluence/statemachine/state/Committer.scala]: Is it necessary?

Comment (DmiSergeev) [reply statemachine/src/main/scala/fluence/statemachine/state/Committer.scala]: It allows separating messages related to different Tendermint blocks.

File: statemachine/src/main/scala/fluence/statemachine/ServerRunner.scala (hunk 2 above)
Comment (DmiSergeev) [reply statemachine/src/main/scala/fluence/statemachine/ServerRunner.scala]: Tendermint changed the default RPC port in one of the recent releases.

File: statemachine/src/main/scala/fluence/statemachine/error/StateMachineError.scala (hunk 4 above)
Comment (DmiSergeev) [reply statemachine/src/main/scala/fluence/statemachine/error/StateMachineError.scala]: Ok

File: statemachine/src/main/scala/fluence/statemachine/ServerRunner.scala (hunk 3 above)
Comment (DmiSergeev) [reply statemachine/src/main/scala/fluence/statemachine/ServerRunner.scala]: Ok

File: docker/README.md (hunk 1 above)
Comment (DmiSergeev) [reply docker/README.md]: Ok

**Pull request merged**